### PR TITLE
HPCC-32118 Default to case-insensitive metric filtering

### PR DESCRIPTION
**Pull request by GordonSmith**
Introduced an option to toggle between case-sensitive and case-insensitive filtering

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [ ] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [ ] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [x] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [x] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [x] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

File: esp/src/src-react/components/Metrics.tsx [hunk 1]
```diff
@@ -67,8 +67,12 @@ class TableEx extends Table {
                 return row[field]?.indexOf && row[field]?.indexOf(filter.substring(colonIdx + 1)) >= 0;
             }
             for (const key in row) {
-                if (row[key]?.indexOf && row[key]?.indexOf(filter) >= 0) {
-                    return true;
+                if (row[key]?.indexOf) {
+                    const value = (!matchCase && row[key]?.toLowerCase) ? row[key]?.toLowerCase() : row[key];
+                    const filterValue = (!matchCase && row[key]?.toLowerCase) ? filter.toLowerCase() : filter;
```
Comment (jeclrsg) [inline esp/src/src-react/components/Metrics.tsx:72]: should this be `!matchCase && filter?.toLowerCase`?

**Review by jeclrsg — CHANGES_REQUESTED** [review]
@GordonSmith One question about a conditional statement. Looked good otherwise

File: esp/src/src-react/components/Metrics.tsx (hunk 1 above)
Comment (GordonSmith) [reply esp/src/src-react/components/Metrics.tsx:72]: No this is correct - the idea is to only lower the filterValue (which is always a string) if the value had been lowered

Comment (GordonSmith) [reply esp/src/src-react/components/Metrics.tsx:72]: Updated logic

File: esp/src/src-react/components/Metrics.tsx [hunk 2]
```diff
@@ -625,7 +629,12 @@ export const Metrics: React.FunctionComponent<MetricsProps> = ({
                 <DockPanel layout={options?.layout} onDockPanelCreate={setDockpanel}>
                     <DockPanelItem key="scopesTable" title={nlsHPCC.Metrics}>
                         <HolyGrail
-                            header={<SearchBox value={scopeFilter} onChange={onChangeScopeFilter} iconProps={filterIcon} placeholder={nlsHPCC.Filter} />}
+                            header={<Stack horizontal>
+                                <Stack.Item grow>
+                                    <SearchBox value={scopeFilter} onChange={onChangeScopeFilter} iconProps={filterIcon} placeholder={nlsHPCC.Filter} />
+                                </Stack.Item>
+                                <ToggleButton appearance="subtle" icon={<TextCaseTitleRegular />} title={nlsHPCC.MatchCase} checked={matchCase} onClick={() => { setMatchCase(!matchCase); }} />
```
Comment (jeclrsg) [inline esp/src/src-react/components/Metrics.tsx:636]: Only thing I noticed when I checked this branch out is the state of this button is pretty subtle. I thought maybe `icon={matchCase ? <TextCaseTitleRegular /> : <TextCaseLowercaseRegular />}`?

**Review by jeclrsg — COMMENTED** [review]
@GordonSmith Only one comment about how that match case button actually shows up in the UI. Seems pretty subtle as to what the status of it is

**Review by jeclrsg — APPROVED** [review]

**Comment by GordonSmith** [comment]
Rebased for pull

**Pull request merged**